msal/src/Microsoft.Identity.Client/Features/ConfidentialClient/ClientCredential.cs

```diff
@@ -47,6 +47,8 @@ public ClientCredential(ClientAssertionCertificate certificate)
         internal ClientAssertionCertificate Certificate { get; private set; }
         internal string Assertion { get; set; }
         internal long ValidTo { get; set; }
+        internal bool CachedX5C { get; set; }
```

Comment:
>CachedX5C [](start = 22, length = 9)

why is this caching introduced? (based on the most recent discussion I thought we wouldn't need it but I may have missed something)